### ISSUE TITLE
Query combined partitioned column and regular column not return correct result

### ISSUE THREAD

**Issue by jeffyecn** [crate/crate#6990]
**CrateDB version**: 2.3.3

**JVM version**: Oracle 1.8.0_161-b12

**OS version / environment description**: Ubuntu 16.04

**Problem description**:

When we have two conditions in the where clause. One is on partition column, another is on regular colum , the condition of partition column was ignored. If only the partition column condition in SQL, it works.

**Steps to reproduce**:

```
CREATE TABLE test_bug (time_id INTEGER, user_id INTEGER, num INTEGER, PRIMARY KEY (time_id, user_id)) PARTITIONED BY (time_id);

INSERT INTO test_bug (time_id, user_id, num) VALUES (1, 1, 1), (2,1,1);

SELECT time_id FROM test_bug WHERE time_id=1 AND num = 1;
```

Got result:
```
+---------+
| time_id |
+---------+
|       2 |
|       1 |
+---------+
```

Expect result:
```
+---------+
| time_id |
+---------+
|       1 |
+---------+
```

**Comment by matriv** [comment]
Confirmed and reproduced. The query "hits" both partitions.

**Comment by jeffyecn** [comment]
Thanks for confirming. 

By any chance you know if v2.3.2 has this problem? The reason I am asking is we need to determine a version to roll back.

Thanks,

**Comment by matriv** [comment]
@jeffyecn Seems that 2.3.2 doesn't have this issue.

**Comment by jeffyecn** [comment]
sounds great! Thanks,

**Comment by matriv** [comment]
@jeffyecn Issue has been fixed and will be available with the next hotfix ``2.3`` release.
